Source/Core/Core/FileMonitor.h

```diff
@@ -0,0 +1,24 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <memory>
+
+#include "Common/CommonTypes.h"
+
+namespace DiscIO
+{
+class IFileSystem;
+class IVolume;
+}
+
+namespace FileMonitor
+{
+// Can be called with nullptr to set the file system to nothing. When not called
+// with nullptr, the volume must remain valid until the next SetFileSystem call.
+void SetFileSystem(const DiscIO::IVolume* volume);
+// SetFileSystem must have been called before calling Log
```

Comment:
Log says that "SetFileSystem" must have been called.  Ensuring that s_filesystem is nullptr would potentially be a way to do that?